lib/circleci/http.rb

```diff
@@ -8,6 +8,7 @@ class Http # @private
 
     def initialize(_config)
       @config, @errors, @success, @over_limit, @suspended = _config, [], false, false, false
+      RestClient.proxy = @config.proxy
```

Comment:
I think we should set the proxy on `RestClient` *only if* it is being configured. So lets add a guard here to configure the proxy only when passed through on the config. I'm aware that RestClient will not do anything if it is set to a *falsey* value but I want to make sure it is an opt-in configuration only and does not affect the general usage if something changes on the RestClient side for whatever reason.